tests/ui/structs/method-chain-expression-failure.stderr

```diff
@@ -0,0 +1,17 @@
+error[E0599]: no method named `foo` found for struct `D` in the current scope
+  --> $DIR/method-chain-expression-failure.rs:24:19
+   |
+LL | struct D;
+   | -------- method `foo` not found for this struct
+...
+LL |     fn foo(&self) {}
+   |     ------------- method available on `&C` here
```

Comment:
@estebank  How do I remove the `&` and the method level comments appear to come ahead (I think since that's the src code order). Is that ok?

Reply:
Ordering is ok, we can side-step that issue by introducing a `note` instead.

There's a `peel_refs` argument on `ty::Ty` IIRC, which would let you do that, but it is fine to leave it as is as well.